source/common/router/scoped_rds.cc

```diff
@@ -308,9 +301,37 @@ void ScopedRdsConfigSubscription::onRdsConfigUpdate(const std::string& scope_nam
 void ScopedRdsConfigSubscription::onConfigUpdate(
     const std::vector<Envoy::Config::DecodedResourceRef>& resources,
     const std::string& version_info) {
+  Protobuf::RepeatedPtrField<std::string> to_remove_repeated;
+  for (const auto& scoped_route : scoped_route_map_) {
+    *to_remove_repeated.Add() = scoped_route.first;
+  }
+  onConfigUpdate(resources, to_remove_repeated, version_info);
+}
+
+Protobuf::RepeatedPtrField<std::string>
```

Comment:
Can you add some comment here explaining what this does? What kind of update conflict?

Reply:
Both scope name and scope key conflicts. The conflict can be between new scopes and old scopes, and it can also be between new scopes.